test/extended/apiserver/openapiv3.go

```diff
@@ -0,0 +1,259 @@
+package apiserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	"github.com/golang/protobuf/proto"
+	openapi_v3 "github.com/google/gnostic/openapiv3"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/kube-openapi/pkg/handler3"
+	"k8s.io/kube-openapi/pkg/spec3"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[sig-api-machinery][Feature:APIServer]", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithoutNamespace("apiserver-openapi")
+
+	g.It("should serve openapi v3 discovery", func() {
+		transport, err := rest.TransportFor(oc.AdminConfig())
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		req, err := http.NewRequest("GET", oc.AdminConfig().Host+"/openapi/v3", nil)
```

Comment:
I wonder does this endpoint open in every openshift environment such as hypershift, microshift, etc.?